### DISCUSSION TITLE
How stable is the internal custom marshaling API in Embind?

### DISCUSSION THREAD

**Discussion by mmarczell-graphisoft** [emscripten-core/emscripten#15834] [Q&A] [answered]
https://github.com/emscripten-core/emscripten/pull/9089 introduced the ability to define custom marshaling for types. The PR comments mention that this is intentionally undocumented and kept in the internal namespace, signifying no commitment to the stability of this API. The API hasn't changed in the 2.5 years since then.

How safe is it to build extensively on this? Are there any current projects, plans etc. that would involve changing this API in the near future?

**Comment by kripken** [comment, marked as answer]
There are some thoughts about doing some work on embind that could change that API. But nothing concrete at the moment, and it's possible another approach will be taken, so it's hard to say how likely such a change is. But it's still true that there is no guarantee about stability there.
